src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -337,6 +357,82 @@ private static void RenderFullDescription(string description, string indent, int
                     maxWidth));
         }
 
+        /// <summary>
+        /// Renders a node in the "ANSI" style.
+        /// </summary>
+        /// <param name="strCollection">
+        /// List of strings to which the node's rendering will be appended.
+        /// </param>
+        /// <param name="indentation">
+        /// The indentation level in chars at which the node should be rendered.
+        /// </param>
+        /// <param name="maxWidth">
+        /// The maximum number of chars that the rendering is allowed to consume.
+        /// </param>
+        private
+        void
+        RenderAnsi(ArrayList strCollection, int indentation, int maxWidth)
+        {
+            string indent = StringUtil.Padding(indentation);
+            string secRemain = string.Empty;
+            if (SecondsRemaining >= 0)
+            {
+                secRemain = SecondsRemaining.ToString() + "s";
+            }
+
+            int secRemainLength = secRemain.Length + 1;
+
+            // limit progress bar to 120 chars as no need to render full width
+            if (maxWidth > 120)
+            {
+                maxWidth = 120;
+            }
```

Comment:
Why have we no need to render full width? Other reneder cases don't use this limitations.